### ISSUE TITLE
Handling conservation laws via solver constraints

### ISSUE THREAD

**Issue by dweindl** [AMICI-dev/AMICI#2329]
The CVODES constraint handling (see also #2327) would also allow us enforce conservation laws, and probably make our code quite a bit simpler.

To be discussed whether we'd want to keep the current conservation law handling anyways in order to support other solver at some point, but it seems rather unlikely that this will happen.

**Comment by FFroehlich** [comment]
Yes getting rid of the conservation laws would indeed be great. Is constraint handling also supported by the non-linear solvers? Moreover, we would have to figure out a way of applying the constraints when computing derivatives implicitly. Not an issue with positivity but more problematic with conservation laws.

**Comment by dweindl** [comment]
> Is constraint handling also supported by the non-linear solvers? 

Yes

> Moreover, we would have to figure out a way of applying the constraints when computing derivatives implicitly.

That I don't know.

**Comment by FFroehlich** [comment]
> > Is constraint handling also supported by the non-linear solvers?
> 
> ~Yes~ Wait, no. Contraints are checked after the non-linear solve.
> 
> > Moreover, we would have to figure out a way of applying the constraints when computing derivatives implicitly.
> 
> That I don't know.

Both would be issues for steadystate solvers.
